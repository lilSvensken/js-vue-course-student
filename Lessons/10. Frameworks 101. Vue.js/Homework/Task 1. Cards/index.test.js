/**
 * @jest-environment jsdom
 */

const { mount } = require("@vue/test-utils");
const Cards = require("./Cards.vue");

describe("10.h.1 Cards", () => {
    const wrapper = mount(Cards.default, {
        data() {
            return {
                cards: [
                    {
                        id: 0,
                        src: '/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/iphone10.jpg',
                        description: 'Packed with Innovative Features Including a Super Retina Display, TrueDepth Camera System, Face ID and A11 Bionic Chip with Neural Engine',
                        head: 'Iphone X 64GB',
                        price: '800$',
                        number: 0,
                        max: 4,
                    },
                    {
                        id: 1,
                        src: '/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/galaxys21.jpg',
                        description: 'Introducing Galaxy S21 Ultra 5G. Designed with a unique contour-cut camera to create a revolution in photography â€” letting you capture cinematic 8K video and snap epic stills, all in one go.',
                        head: 'Samsung Galaxy S21 128GB',
                        price: '1000$',
                        number: 0,
                        max: 10,
                    },
                    {
                        id: 2,
                        src: '/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/nokia2600.jpg',
                        description: 'Give me back my 2007',
                        head: 'Nokia 2600',
                        price: 'Priceless',
                        number: 0,
                        max: 1,
                    },
                ],
            };
        },
    });
    expect(wrapper.element).toMatchSnapshot();

    it("10.h.1.1 Cards has data", () => {
        expect(typeof Cards.default.data).toBe("function");
    });

    const plusButtons = wrapper.findAll('.button-plus');
    const minusButtons = wrapper.findAll('.button-minus');

    it("10.h.1.3 first card plus button click", async () => {
        const plusButtons = wrapper.findAll('.button-plus');
        const minusButtons = wrapper.findAll('.button-minus');
        const firstCardPlusButton = plusButtons.at(0);

        await firstCardPlusButton.trigger("click");
        expect(wrapper.element).toMatchSnapshot();
    });

    it("10.h.1.4 second card minus button click", async () => {
        const secondCardMinusButton = minusButtons.at(1);

        await secondCardMinusButton.trigger("click");
        expect(wrapper.element).toMatchSnapshot();
    });

    it("10.h.1.5 third card plus button two clicks", async () => {
        const thirdCardPlusButton = plusButtons.at(2);

        await thirdCardPlusButton.trigger("click");
        await thirdCardPlusButton.trigger("click");
        expect(wrapper.element).toMatchSnapshot();
    });

    it("10.h.1.6 third card minus button click", async () => {
        const thirdCardMinusButton = minusButtons.at(2);

        await thirdCardMinusButton.trigger("click");
        expect(wrapper.element).toMatchSnapshot();
    });

    it("10.h.1.7 third card minus button two more clicks", async () => {
        const thirdCardMinusButton = minusButtons.at(2);

        await thirdCardMinusButton.trigger("click");
        await thirdCardMinusButton.trigger("click");
        expect(wrapper.element).toMatchSnapshot();
    });
});
