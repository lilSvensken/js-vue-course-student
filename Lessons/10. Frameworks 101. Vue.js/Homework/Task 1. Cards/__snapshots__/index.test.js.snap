// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  class="card-container"
>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/iphone10.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Iphone X 64GB
      </h5>
       
      <h4
        class="card-title"
      >
        800$
      </h4>
       
      <p
        class="card-text"
      >
        Packed with Innovative Features Including a Super Retina Display, TrueDepth Camera System, Face ID and A11 Bionic Chip with Neural Engine
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          0 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/galaxys21.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Samsung Galaxy S21 128GB
      </h5>
       
      <h4
        class="card-title"
      >
        1000$
      </h4>
       
      <p
        class="card-text"
      >
        Introducing Galaxy S21 Ultra 5G. Designed with a unique contour-cut camera to create a revolution in photography — letting you capture cinematic 8K video and snap epic stills, all in one go.
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          0 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/nokia2600.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Nokia 2600
      </h5>
       
      <h4
        class="card-title"
      >
        Priceless
      </h4>
       
      <p
        class="card-text"
      >
        Give me back my 2007
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          0 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`10.h.1 Cards 10.h.1.3 first card plus button click 1`] = `
<div
  class="card-container"
>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/iphone10.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Iphone X 64GB
      </h5>
       
      <h4
        class="card-title"
      >
        800$
      </h4>
       
      <p
        class="card-text"
      >
        Packed with Innovative Features Including a Super Retina Display, TrueDepth Camera System, Face ID and A11 Bionic Chip with Neural Engine
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          1 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/galaxys21.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Samsung Galaxy S21 128GB
      </h5>
       
      <h4
        class="card-title"
      >
        1000$
      </h4>
       
      <p
        class="card-text"
      >
        Introducing Galaxy S21 Ultra 5G. Designed with a unique contour-cut camera to create a revolution in photography — letting you capture cinematic 8K video and snap epic stills, all in one go.
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          0 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/nokia2600.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Nokia 2600
      </h5>
       
      <h4
        class="card-title"
      >
        Priceless
      </h4>
       
      <p
        class="card-text"
      >
        Give me back my 2007
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          0 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`10.h.1 Cards 10.h.1.4 second card minus button click 1`] = `
<div
  class="card-container"
>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/iphone10.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Iphone X 64GB
      </h5>
       
      <h4
        class="card-title"
      >
        800$
      </h4>
       
      <p
        class="card-text"
      >
        Packed with Innovative Features Including a Super Retina Display, TrueDepth Camera System, Face ID and A11 Bionic Chip with Neural Engine
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          1 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/galaxys21.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Samsung Galaxy S21 128GB
      </h5>
       
      <h4
        class="card-title"
      >
        1000$
      </h4>
       
      <p
        class="card-text"
      >
        Introducing Galaxy S21 Ultra 5G. Designed with a unique contour-cut camera to create a revolution in photography — letting you capture cinematic 8K video and snap epic stills, all in one go.
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          0 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/nokia2600.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Nokia 2600
      </h5>
       
      <h4
        class="card-title"
      >
        Priceless
      </h4>
       
      <p
        class="card-text"
      >
        Give me back my 2007
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          0 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`10.h.1 Cards 10.h.1.5 third card plus button two clicks 1`] = `
<div
  class="card-container"
>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/iphone10.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Iphone X 64GB
      </h5>
       
      <h4
        class="card-title"
      >
        800$
      </h4>
       
      <p
        class="card-text"
      >
        Packed with Innovative Features Including a Super Retina Display, TrueDepth Camera System, Face ID and A11 Bionic Chip with Neural Engine
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          1 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/galaxys21.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Samsung Galaxy S21 128GB
      </h5>
       
      <h4
        class="card-title"
      >
        1000$
      </h4>
       
      <p
        class="card-text"
      >
        Introducing Galaxy S21 Ultra 5G. Designed with a unique contour-cut camera to create a revolution in photography — letting you capture cinematic 8K video and snap epic stills, all in one go.
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          0 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/nokia2600.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Nokia 2600
      </h5>
       
      <h4
        class="card-title"
      >
        Priceless
      </h4>
       
      <p
        class="card-text"
      >
        Give me back my 2007
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          1 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`10.h.1 Cards 10.h.1.6 third card minus button click 1`] = `
<div
  class="card-container"
>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/iphone10.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Iphone X 64GB
      </h5>
       
      <h4
        class="card-title"
      >
        800$
      </h4>
       
      <p
        class="card-text"
      >
        Packed with Innovative Features Including a Super Retina Display, TrueDepth Camera System, Face ID and A11 Bionic Chip with Neural Engine
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          1 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/galaxys21.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Samsung Galaxy S21 128GB
      </h5>
       
      <h4
        class="card-title"
      >
        1000$
      </h4>
       
      <p
        class="card-text"
      >
        Introducing Galaxy S21 Ultra 5G. Designed with a unique contour-cut camera to create a revolution in photography — letting you capture cinematic 8K video and snap epic stills, all in one go.
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          0 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/nokia2600.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Nokia 2600
      </h5>
       
      <h4
        class="card-title"
      >
        Priceless
      </h4>
       
      <p
        class="card-text"
      >
        Give me back my 2007
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          0 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
</div>
`;

exports[`10.h.1 Cards 10.h.1.7 third card minus button two more clicks 1`] = `
<div
  class="card-container"
>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/iphone10.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Iphone X 64GB
      </h5>
       
      <h4
        class="card-title"
      >
        800$
      </h4>
       
      <p
        class="card-text"
      >
        Packed with Innovative Features Including a Super Retina Display, TrueDepth Camera System, Face ID and A11 Bionic Chip with Neural Engine
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          1 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/galaxys21.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Samsung Galaxy S21 128GB
      </h5>
       
      <h4
        class="card-title"
      >
        1000$
      </h4>
       
      <p
        class="card-text"
      >
        Introducing Galaxy S21 Ultra 5G. Designed with a unique contour-cut camera to create a revolution in photography — letting you capture cinematic 8K video and snap epic stills, all in one go.
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          0 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
  <div
    class="card"
  >
    <img
      class="card-img-top"
      src="/Lessons/9. Frameworks 101. Vue.js/Homework/Task 1. Cards/assets/nokia2600.jpg"
    />
     
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Nokia 2600
      </h5>
       
      <h4
        class="card-title"
      >
        Priceless
      </h4>
       
      <p
        class="card-text"
      >
        Give me back my 2007
      </p>
       
      <div
        class="card__buttons"
      >
        <button
          class="btn btn-primary button-plus"
        >
          +
        </button>
         
        <button
          class="btn btn-primary button-minus"
        >
          -
        </button>
         
        <div>
          0 шт.
        </div>
      </div>
       
      <!---->
    </div>
  </div>
</div>
`;
